Source/ARTTypes.m

```diff
@@ -387,20 +387,25 @@ - (nullable NSData *)art_archive {
 #endif
 }
 
-+ (nullable id)art_unarchive:(NSData *)data {
++ (nullable id)art_unarchiveWithAllowedClass:(Class)cls fromData:(NSData *)data {
+    if (data == nil) {
```

Comment:
I'm wondering why you introduced this check – `data` is not declared as nullable. Did you see something that led you to believe it might be `nil` sometimes?

Reply:
It's just migrated from older function, I didn't pay attention to nil checks before recent issues with them. Thanks for pointing out, will fix it.

Reply:
@lawrence-forooghian

Reply:
I think it's good that we try and fix this warning about `data` being `nil`. But my worry is that the check you've added here is covering up a larger problem. Because, we have a non-nullable `data` parameter here, which we've observed is actually sometimes `nil` at runtime. So this method's declaration is incorrect and needs to be fixed. And similarly for the declarations in the methods that call this method, e.g. `+[ARTPushActivationState unarchive:]`, whose argument is also declared as non-nullable. And possibly the methods that call that method and so on.

So I'd suggest we keep the scope of this PR limited to fixing the warning about not passing a class to the unarchiver, and create a separate issue for fixing the nullability stuff. What do you think?

Reply:
Yes, correct.

Reply:
Fixed b28c7c2